patches/CreatePBP.cmake

```diff
@@ -0,0 +1,55 @@
+# File defining macro outputting PSP-specific EBOOT.PBP out of passed executable target.
+# Copyright 2020 - Daniel 'dbeef' Zalega
+#
+# Args:
+# target - defined by an add_executable call before calling create_pbp_file
+# icon_path - path to .png file, 144x82, pass "NULL" to use default
+# background_path - path to .png file, 480x272, pass "NULL" to use default
+# preview_path - path to .png file, 480x272, pass "NULL" to use default
+
+cmake_minimum_required(VERSION 3.10)
+
+macro(create_pbp_file target icon_path background_path preview_path)
+
+    add_custom_command(TARGET ${target} POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E make_directory
+            $<TARGET_FILE_DIR:${target}>/psp_artifact
+            COMMENT "Creating psp_artifact directory."
+            )
+
+    add_custom_command(TARGET ${target} POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E copy_if_different
+            $<TARGET_FILE:${target}>
+            $<TARGET_FILE_DIR:${target}>/psp_artifact
+            COMMENT "Copying ELF to psp_arfitact directory."
+            )
+
+    add_custom_command(
+            TARGET ${target}
+            POST_BUILD COMMAND
+            "psp-fixup-imports" "$<TARGET_FILE_DIR:${target}>/psp_artifact/${target}"
+            COMMENT "Calling psp-fixup-imports"
+    )
+
+    add_custom_command(
+            TARGET ${target}
+            POST_BUILD COMMAND
+            "mksfoex" "-d" "MEMSIZE=1" "$<TARGET_FILE_DIR:${target}>/psp_artifact/${target}" "PARAM.SFO"
+            COMMENT "Calling mksfoex"
+    )
+
+    add_custom_command(
+            TARGET ${target}
+            POST_BUILD COMMAND
+            "pack-pbp" "EBOOT.PBP" "PARAM.SFO" "${icon_path}" "NULL" "${preview_path}"
+            "${background_path}" "NULL" "$<TARGET_FILE_DIR:${target}>/psp_artifact/${target}" "NULL"
```

Comment:
Same here, one might want to specify some assets, while leaving out others.
We should make them all available, but defaulting to `NULL`

Reply:
Will provide option to pass the missing arguments.

Meanwhile, could you please help me with naming them?
Calling pack-pbp does not provide me any clue about what  "snd0.at3", "data.psp" or "data.psar" mean, nor have I ever used them.

```
dbeef@dbeefstation:~$ pack-pbp
USAGE: pack-pbp <output.pbp> <param.sfo> <icon0.png> <icon1.pmf> <pic0.png> <pic1.png> <snd0.at3> <data.psp> <data.psar>
```

Reply:
After some conversations, now I know that `snd0.at3` is an ATRAC encoded audio file.

Reply:
We should add some documentation to the tools, right.
- `data.psp` is the executable (elf/prx)
- `data.psar` can contain an encrypted iso file for firmware updates and such

Reply:
I changed this macro to take optional arguments, see relevant commit:

https://github.com/pspdev/psplibraries/pull/52/commits/473ad130040f410eb71a511223250529dc4813b0

Now macro can be called simply by:
```
create_pbp_file(TARGET test-cmake)
```

However, as these leftover pack-pbp arguments are not well documented and I hardly find any use for them, I propose not adding them only for the sake of completeness.